### PR TITLE
Dynamic Filtering

### PR DESCRIPTION
**Pull request by enisn**
## Summary
This PR includes dynamic filtering feature. For example: generating a LINQ query from a Dictionary like 
```
{ "TotalPage.gte", "5" },
{ "Title.contains", "a" }
```
Following query will be generated:
```csharp
...Where(x => x.TotalPage >= 5 && x.Title.Contains("a"))
```

### CODE REVIEW

File: AutoFilterer.Dynamics/DynamicFilter.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                    var exp = new OperatorComparisonAttribute(OperatorType.Equal).BuildExpression(body, targetProperty, filterProperty: null, value);
+
+                    var combined = Combine(finalExpression, exp);
+                    finalExpression = Combine(body, combined);
+                }
+                else
+                {
+                    var splitted = key.Split('.');
+                    if (splitted.Length == 2)
+                    {
```
Comment (furkandeveloper) [inline AutoFilterer.Dynamics/DynamicFilter.cs:82]: Can be recursive function? @enisn

**Comment by furkandeveloper** [comment]
Against git flow. `feature/*` branches not must merge master branch. @enisn

File: AutoFilterer.Dynamics/DynamicFilter.cs (hunk 1 above)
Comment (enisn) [reply AutoFilterer.Dynamics/DynamicFilter.cs]: This is a temporary solution until inner objects implemented.
I've just made that check as method to better understanding what does it check.

You can check Commit 000c8328

**Review by furkandeveloper — APPROVED** [review]
Approved

**Comment by enisn** [comment]
Thank you for your review @furkandeveloper

**Pull request merged**